src/Mvc/Mvc.Core/test/ModelBinding/Metadata/ModelAttributesTest.cs

```diff
@@ -287,6 +287,18 @@ public void GetAttributesForProperty_WithModelType_IncludesTypeAttributes()
             attribute => Assert.IsType<ClassValidator>(attribute));
     }
 
+    [Fact]
+    public void GetAttributeForProperty_WithModelType_HandlesMultipleAttributesOnType()
```

Comment:
Great!

This covers ModelMetadataType. Are there other attributes that had AllowMultiple=false?

Reply:
`ModelBinder` has it, but it's behavior is a little interesting. When constructing the model-binding context, we capture all the attributes on a type, including inherited ones ([ref](https://source.dot.net/#Microsoft.AspNetCore.Mvc.Core/ApplicationModels/DefaultApplicationModelProvider.cs,220)). When constructing the actual model binding context, we pick the attribute closest to the type were evaluating and use its properties to construct the context ([ref](https://source.dot.net/#Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/BindingInfo.cs,120)).

So unlike the `ModelMetadata` scenario, there isn't a place in code where we try to evaluate a single attribute on a type. We could modify the binding logic above to throw if there are multiple attributes on the same type. That would introduce a new exception in the code (although it's not possible to hit that exception without this current change).

Ditto the same situation for `ProducesAttribute`. Although we only expect one attribute on a particular type, all the codepaths that consume it assume that attributes can be inherited and don't strictly expect a single attribute to be defined.

So we _can_ through exceptions at runtime that warn about multiple attributes on the same type but that feels a little heavy handed given the existing behavior for these attributes. 

Thoughts?